modules/dxp/apps/saml/saml-impl/src/main/java/com/liferay/saml/internal/servlet/filter/SpSessionTerminationSamlPortalFilter.java

```diff
@@ -73,8 +75,14 @@ protected void doProcessFilter(
 			_singleLogoutProfile.terminateSpSession(
 				httpServletRequest, httpServletResponse);
 
-			_singleLogoutProfile.logout(
-				httpServletRequest, httpServletResponse);
+			SamlIdpSpSession samlIdpSpSession =
+				_samlIdpSpSessionLocalService.fetchSamlIdpSpSession(
+					samlSpSession.getSamlSpSessionId());
```

Comment:
The `SamlIdpSpSessionLocalService` manages data for the SAML IDP. This filter is executing for SP requests, so we cannot call that here.